### PR TITLE
feat: modules enhancement 

### PR DESCRIPTION
**Pull request by VGabriel45**
## PR-Codex overview
This PR focuses on refactoring the `OwnableValidator` and `K1Validator` modules, enhancing their functionality, and updating related tests to improve code quality and maintainability.

### Detailed summary
- Deleted several unused module files.
- Updated `package.json` dependencies.
- Made `parameters` optional in several function signatures.
- Renamed `account` variable to `eoaAccount` for clarity.
- Enhanced the `addOwner`, `removeOwner`, and `setThreshold` functions in `OwnableValidator`.
- Improved test cases for `OwnableValidator` and `K1Validator` to reflect new changes.
- Updated `toNexusAccount` to support new validator modules.
- Ensured proper handling of signatures in user operations.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -125,6 +125,7 @@
     "commit-msg": "npx --no -- commitlint --edit ${1}"
   },
   "dependencies": {
+    "@rhinestone/module-sdk": "^0.1.17",
     "merkletreejs": "^0.3.11"
```
Comment (joepegler) [inline package.json:129]: Can we remove merkletree?

File: src/sdk/clients/createNexusClient.test.ts [hunk 2]
```diff
@@ -157,19 +150,19 @@ describe("nexus.client", async () => {
     expect(result).toBeTruthy()
   }, 30000)
 
-  test.skip("should create a nexusAccount from an ethers signer", async () => {
-    const ethersProvider = new JsonRpcProvider(chain.rpcUrls.default.http[0])
-    const ethersSigner = new Wallet(pKey, ethersProvider)
+  // test.skip("should create a nexusAccount from an ethers signer", async () => {
```
Comment (joepegler) [inline src/sdk/clients/createNexusClient.test.ts:153]: Let's just remove comments

File: src/sdk/clients/createNexusClient.ts [hunk 3]
```diff
@@ -212,6 +211,8 @@ export async function createNexusClient(
     key,
     name,
     account: nexusAccount,
+    paymaster,
```
Comment (joepegler) [inline src/sdk/clients/createNexusClient.ts:214]: Remove these. I think you're removing some changes that I've made by accident somehow.

File: src/sdk/clients/decorators/erc7579/erc7579.decorators.test.ts [hunk 4]
```diff
@@ -31,13 +40,13 @@ describe("erc7579.decorators", async () => {
 
     chain = network.chain
     bundlerUrl = network.bundlerUrl
-    account = getTestAccount(0)
+    eoaAccount = getTestAccount(0)
```
Comment (joepegler) [inline src/sdk/clients/decorators/erc7579/erc7579.decorators.test.ts:43]: Replace account everywhere then, but I thought account was fine.

File: src/sdk/clients/decorators/erc7579/erc7579.decorators.test.ts [hunk 5]
```diff
@@ -85,28 +94,42 @@ describe("erc7579.decorators", async () => {
...
 
     const { success } = await nexusClient.waitForUserOperationReceipt({ hash })
     expect(success).toBe(true)
   })
 
-  test.skip("should install a module", async () => {
-    const hash = await nexusClient.installModule({
+  test("should uninstall a module", async () => {
+    const [installedValidators, nextCursor] =
+      await nexusClient.getInstalledValidators({})
```
Comment (joepegler) [inline src/sdk/clients/decorators/erc7579/erc7579.decorators.test.ts:112]: Can you make this work without the empty bracket? I briefly tried and couldn't figure it out.

File: src/sdk/clients/decorators/erc7579/getPreviousModule.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+): Promise<Hex> {
+  const { account: account_ = client.account, module } = parameters
+
+  if (!account_) {
+    throw new AccountNotFoundError({
+      docsPath: "/docs/actions/wallet/sendTransaction"
+    })
+  }
+
+  console.log(parameters, "parameters")
```
Comment (joepegler) [inline src/sdk/clients/decorators/erc7579/getPreviousModule.ts:57]: Log

File: src/sdk/modules/validators/ownableValidator/ownableValidator.test.ts [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+      transport: http(),
+      bundlerTransport: http(bundlerUrl)
+    })
+
+    nexusAccountAddress = await nexusClient.account.getCounterFactualAddress()
+    await fundAndDeployClients(testClient, [nexusClient])
+
+    ownableValidatorModule = await toOwnableValidatorModule({
+      address: TEST_CONTRACTS.OwnableValidator.address,
+      client: nexusClient.account.client as PublicClient,
```
Comment (joepegler) [inline src/sdk/modules/validators/ownableValidator/ownableValidator.test.ts:78]: I don't like that we make the user do this. Can we add a 'toInitData' / getInitData helper for each module? This way we keep the decorators mostly account agnostic

File: src/sdk/modules/validators/ownableValidator/toOwnableValidationModule.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        message: { raw: userOpHash as Hex }
+      })
+      return signature as Hex
+    },
+    signMessage: async (_message: Uint8Array | string) => {
+      const message =
+        typeof _message === "string" ? _message : { raw: _message }
+      let signature = await signer.signMessage({ message })
+
+      const potentiallyIncorrectV = Number.parseInt(signature.slice(-2), 16)
```
Comment (joepegler) [inline src/sdk/modules/validators/ownableValidator/toOwnableValidationModule.ts:46]: Let's stick this function somewhere and just reuse it.

**Comment by joepegler** [comment]
Conflicts

File: src/sdk/clients/createNexusClient.ts (hunk 3 above)
Comment (VGabriel45) [reply src/sdk/clients/createNexusClient.ts:214]: Remove what exactly ? The "paymaster" ?

File: src/sdk/clients/decorators/erc7579/erc7579.decorators.test.ts (hunk 4 above)
Comment (VGabriel45) [reply src/sdk/clients/decorators/erc7579/erc7579.decorators.test.ts:43]: account gets confused with smart account which is also named "account" in "nexusClient.account", will replace everywhere

**Comment by joepegler** [comment]
Can you convince me that the best devEx for people wouldn’t be to add decorators that apply to their smart account here: https://github.com/bcnmy/sdk/blob/develop/src/sdk/clients/createNexusClient.ts#L222. For example, I would expect the following actions to be exportable from our modules folder:

```
nexusClient
.extend(smartSessionActions())
.extend(ownableActions())
.extend(passkeyActions())

```

File: src/sdk/clients/decorators/erc7579/erc7579.decorators.test.ts (hunk 5 above)
Comment (VGabriel45) [reply src/sdk/clients/decorators/erc7579/erc7579.decorators.test.ts:112]: Fixed

File: src/sdk/modules/validators/ownableValidator/ownableValidator.test.ts (hunk 7 above)
Comment (VGabriel45) [reply src/sdk/modules/validators/ownableValidator/ownableValidator.test.ts:76]: Yea sure, will fix this in another PR, there are other improvements we can do as well.

File: src/sdk/modules/validators/ownableValidator/toOwnableValidationModule.ts (hunk 8 above)
Comment (VGabriel45) [reply src/sdk/modules/validators/ownableValidator/toOwnableValidationModule.ts]: Fixed

**Comment by VGabriel45** [comment]
> Can you convince me that the best devEx for people wouldn’t be to add decorators that apply to their smart account here: https://github.com/bcnmy/sdk/blob/develop/src/sdk/clients/createNexusClient.ts#L222. For example, I would expect the following actions to be exportable from our modules folder:
> 
> ```
> nexusClient
> .extend(smartSessionActions())
> .extend(ownableActions())
> .extend(passkeyActions())
> ```

I don't want to convince you since it sounds good xD. However we still need toK1ValidatorModule to create instances we can use on the smart account because we need custom behaviour for getDummySignature, signUserOperation or we might need extra methods like getSessionId ... which are required to be called inside the smart account client instance, before it's created.

I do agree with your point for actions.

File: src/sdk/modules/validators/ownableValidator/ownableValidator.test.ts (hunk 7 above)
Comment (joepegler) [reply src/sdk/modules/validators/ownableValidator/ownableValidator.test.ts:76]: Awesome 😎

File: src/sdk/clients/createNexusClient.ts (hunk 3 above)
Comment (joepegler) [reply src/sdk/clients/createNexusClient.ts]: I meant the line below this, sorry

File: src/sdk/clients/decorators/erc7579/modules/ownableValidator/addOwner.ts [hunk 9]
```diff
@@ -0,0 +1,86 @@
+import { getAddOwnableValidatorOwnerAction } from "@rhinestone/module-sdk"
```
Comment (joepegler) [inline src/sdk/clients/decorators/erc7579/modules/ownableValidator/addOwner.ts:1]: Can you move this whole decorators folder into `modules/validators/ownableValidator?` I think it makes more sense there.

**Pull request merged**